config/components/manager/manager.yaml

```diff
@@ -49,4 +49,9 @@ spec:
             cpu: 500m
             memory: 512Mi
       serviceAccountName: controller-manager
+      tolerations:
+        - key: kueue.x-k8s.io/nodegroup
```

Comment:
Could you explain location where we use this node tain?

Reply:
thanks for taking a look.
The idea is to allow user to have a dedicated nodepool for the controller components, and promote isolation, stability and maintenance easiness.

Reply:
I see. In that case, could you update the documentation how users can this toleration and taints?
I think we can expand the following documentation: https://kueue.sigs.k8s.io/docs/installation/#install-a-released-version

Reply:
i see, thanks for explaining. So the idea is that an admin creates a  node group with this taint so that kueue manager does not get preempted. 

I'm wondering yet if the toleration is enough to prefer the nodes in case there is also enough resources on other nodes? Maybe we should also add preferredDuringSchedulingIgnoredDuringExecution? wdyt @tenzen-y?

cc @gabesaba was doing similar setup to isolate the kueue manager.

Reply:
do you mean to have affinity in addition of toleration? and configure the node to have both taint and label?

Reply:
Yeah, but not sure what are the best practices here. Maybe @alculquicondor can suggest also the approach.

Reply:
> I'm wondering yet if the toleration is enough to prefer the nodes in case there is also enough resources on other nodes? Maybe we should also add preferredDuringSchedulingIgnoredDuringExecution? wdyt @tenzen-y?

That makes sense. If we want to add tolerations, we should add affinity as well. 
As an alternative solution to address motivation, we can provide the dedicated priority. But, the priority depends on the each cluster. So, the taints and affinity would be better.

One concern is users need to obtain the permissions which can be modified the taints and node labels.
And, I know some cloud provider restrict to access to Node, directly.

So, I would recommend to mention the permission limitations when batch admins want to use the Kueue nodepool feature.

Reply:
@alculquicondor Agree user can always modify the manifest to suit their need, though I think adding it here could bring value because we are promoting best practice to improve the isolation, stability of the controller.

@tenzen-y to your point about permission to update the node. I believe in most case kueue and other components are installed and configured by admin, who should also have admin access to the cluster. But nevertheless, I think toleration and preferredDuringSchedulingIgnoredDuringExecution is flexible enough to schedule the controller to nodes that doesn't meet the criteria. 

Thanks all for the input and open to suggestion.

Reply:
> @tenzen-y to your point about permission to update the node. I believe in most case kueue and other components are installed and configured by admin, who should also have admin access to the cluster.

Yes, that's right. But AFAIK, the GKE autopilot have limitation to modify nodeLabels, directly. Instead of modifying the nodeLabels by kubectrl, they need to use the gcloud or something. In some cases, cluster admins and batch admins have different personas.

Reply:
I would find it unlikely that the person/team who maintains the Kueue installation isn't also the cluster admin who creates the nodes.

>  though I think adding it here could bring value because we are promoting best practice to improve the isolation, stability of the controller.

I would prefer to document this rather than adding a fixed toleration. I imagine a lot of cluster admins already use a single taint for all their control-plane pods and they probably would just use that instead of using one provided by the kueue repo.

Reply:
While not opposed to the idea of embedding `toleration` + `preferredDuringSchedulingIgnoredDuringExecution` I think we can start with documentation as a less committal solution.

As noted before, to benefit from `toleration` + `preferredDuringSchedulingIgnoredDuringExecution` an admin still needs to:
1. know about the feature as this is not a standard practice, 
2. label and taint the nodes accordingly

IMO (1.) and (2.) would still warrant docs on this in the tasks page. And it would be ok to add an extra step about adding `toleration` + `preferredDuringSchedulingIgnoredDuringExecution` (or even `requiredDuringSchedulingIgnoredDuringExecution`).

Reply:
+1 on having a page for this, potentially named "Productionisation". Other than the topic of putting kueue in dedicated nodes, we can mention how to setup requests

Reply:
Opened an issue for this: https://github.com/kubernetes-sigs/kueue/issues/3087, leaving it open if single page or two pages.

Reply:
+1
Thanks.